DTS-STN/dts-rmp
# 115 engagement edit

# Description

[#115 View Engagement: Edit mode](https://taiga.dts-stn.com/project/pond-knowlege-management-portal/us/115?milestone=20)

This PR contains the feature for editing engagements.

# Story Acceptance Criteria

[#115 View Engagement: Edit Mode](https://taiga.dts-stn.com/project/pond-knowlege-management-portal/us/115?milestone=20)

As a policy analyst I want to be able to update Engagement details so that RMP users always have up to date info on Engagements in the system.

Acceptance Criteria:

- User performs Engagement search and clicks a Engagement from the search results, Engagement detail view page is displayed and all Engagement information is displayed
- User clicks "Edit Engagement" button and Engagement detail page changes state to edit mode displaying user inputs to change engagement info, displaying Cancel/Submit buttons
- User updates Engagement fields with valid user input and clicks submit, Engagement form is hidden and Engagement detail page is displayed with updated info and success prompt is displayed(toast message top of page? match strategy with Contact detail view)
- User updates Engagement fields with invalid user input and clicks submit, error prompt is displayed, informing user of form validation errors
- User clicks "Cancel" from the Engagement card:edit mode and returns to Engagement detail view
- Above passes on Web/Mobile
- All Engagement data fields are displayed and can be edited, excluding comments

Exclusions:

- Comments submitted when creating the engagement are not editable

## Test Instructions

1. Go to Engagement detail view
1. Select Edit at the bottom of the page
1. See edit page is populated with the data of the current Engagement
1. Modify fields with valid and invalid values, see validation
1. Save engagement with invalid fields, validation behaviour should occur
1. Save engagement with valid fields, you should be returned to the detail page which should have the modified fields and a success notification should appear at the top of the page
1. If there is an error with the db, user should remain on the page and see a notification for the error

## Checklist
